### DISCUSSION TITLE
Is non-string exception codes supported?

### DISCUSSION THREAD

**Discussion by andrew-demb** [vimeo/psalm#7846] [Q&A] [answered]
PHP allows specifying string values to exception codes: https://3v4l.org/mgE0M (I know, it is a bit of a "hack")

But psalm emits an error about possible return type from `Exception::getCode()` only as `int`: https://psalm.dev/r/0cc72947c2

> ERROR: [TypeDoesNotContainType](https://psalm.dev/056) - 14:10 - int cannot be identical to 'custom_code'

Is it intended by psalm - narrow down possible return types to `int`?

**Comment by orklah** [comment, marked as answer]
It is indeed intended. There were a lot of [discussions about this](https://github.com/vimeo/psalm/issues/4295) and specifically about your case [here](https://github.com/vimeo/psalm/issues/4295#issuecomment-914004361) because there is no good solution for every case. The [current compromise](https://github.com/vimeo/psalm/pull/7390) is already pretty complex but it covers a lot of use cases and is consistent with the behaviour in PHPStan so I don't think we'll change this in the core

If it's really important for you, there may be a way to make a plugin or a stub in order to change Psalm's behaviour, but I wouldn't recommend it

**Reply by andrew-demb** [reply]
Thank you for your answer. I will suppress issues and plan to refactor these cases
